autoblocks/_impl/tracer.py

```diff
@@ -27,6 +31,28 @@
 
 evaluator_semaphore_registry: dict[str, asyncio.Semaphore] = {}  # evaluator_id -> semaphore
 
+background_tasks: Set[Union[asyncio.Task[Any], Future[Any]]] = set()
+
+
+@atexit.register
```

Comment:
this is not doing anything since we're still calling `result()` so should always be empty on exit in theory, but wondering if this approach will work in general (as opposed to a queue)

Reply:
ya i was wondering the same, wasn't sure exactly all you and dan had tried in the other approach.

Reply:
I think we got stuck on keeping the daemon thread alive on shutdown iirc, might run into the same issue here